### PR TITLE
SIMS #163 - Student Confirms Consent to Share and Validate Income with CRA

### PR DESCRIPTION
**Pull request by andrewsignori-aot**
Added CRA consent checkbox with the consent text, as shown below.

image

image

### CODE REVIEW

File: sources/packages/web/src/components/fa-application/Section.vue [hunk 1]
```diff
@@ -17,10 +19,37 @@
...
+}
+
 export default {
   props: {
     title: String,
     subTitle: String,
+    type: {
+      type: String,
+      default: StyleTypes.primary,
+      validator: (prop: StyleTypes) => Object.values(StyleTypes).includes(prop),
```
Comment (2011sandeepraj) [inline sources/packages/web/src/components/fa-application/Section.vue:37]: Not clear where is this being used

File: sources/packages/web/src/components/fa-application/Section.vue [hunk 2]
```diff
@@ -17,10 +19,37 @@
...
+      switch (props.type) {
+        case StyleTypes.secondary:
+          return "fa-section-header-secondary";
+        case StyleTypes.additional:
+          return "fa-section-header-additional";
+        default:
+          return "fa-section-header";
+      }
+    });
+
```
Comment (2011sandeepraj) [inline sources/packages/web/src/components/fa-application/Section.vue:51]: This is very good

File: sources/packages/web/src/components/fa-application/Section.vue [hunk 3]
```diff
@@ -1,7 +1,9 @@
 <template>
   <div class="p-component p-mx-4">
     <div class="p-grid">
```
Comment (2011sandeepraj) [inline sources/packages/web/src/components/fa-application/Section.vue:3]: Looks like this component existed before this PR and you modified it. But can you add a comment somewhere to this file to say in what scenarios should some page use this component

File: sources/packages/web/src/views/FinancialInfo.vue [hunk 4]
```diff
@@ -3,7 +3,10 @@
     <div class="p-card p-m-4">
       <Section title="More questions to come..." />
       <ContentGroup>
-        <Section :title="questions.previousYearTaxReturn.title">
+        <Section
+          type="secondary"
+          :title="questions.previousYearTaxReturn.title"
+        >
           <template #sub-title>
             <StringToHtmlParagraphs
```
Comment (2011sandeepraj) [inline sources/packages/web/src/views/FinancialInfo.vue:11]: Missed to mention this in the review of previous PR, but please add a comment somewhere in the StringToHTMLParagraphs.vue to mention when should someone use this component and what will page look like when this used. Eg. "///Long text or String// wil look like <This>

File: sources/packages/web/src/components/ReadMoreContainer.vue [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    const extendedLabel = ref(props.collapsedText);
+
+    const toggleExtended = () => {
+      isExtended.value = !isExtended.value;
+      extendedLabel.value = isExtended.value
+        ? props.expandedText
+        : props.collapsedText;
+    };
+
+    return { isExtended, extendedLabel, toggleExtended };
```
Comment (2011sandeepraj) [inline sources/packages/web/src/components/ReadMoreContainer.vue:39]: Some small documentation, on when to use and why to use this component and result of use will help. Infact, i think we should create a README of all components in the long run to document the usage with screenshots. However, for now, a comment - a line or 2 explaining what will this component help to achieve will suffice

Comment (2011sandeepraj) [reply sources/packages/web/src/components/ReadMoreContainer.vue:39]: For this the comment could say something like "This component can be used wherever there are a long/lot of texts to be read. This component provides a Read More/Read Less options on the page"

File: sources/packages/web/src/components/ReadMoreContainer.vue [hunk 6]
```diff
@@ -0,0 +1,42 @@
+<template>
+  <div>
+    <slot></slot>
+    <div id="extendedContent" v-show="isExtended">
+      <slot name="extended"></slot>
+    </div>
+    <a id="toggleButton" @click.prevent="toggleExtended" href="#">{{
```
Comment (2011sandeepraj) [inline sources/packages/web/src/components/ReadMoreContainer.vue:7]: is providing href=# mandatory?

File: sources/packages/web/src/components/ReadMoreContainer.vue [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    },
+    expandedText: {
+      type: String,
+      default: "Read Less",
+      required: true,
+    },
+  },
+  setup(props: any) {
+    const isExtended = ref(false);
+    const extendedLabel = ref(props.collapsedText);
```
Comment (2011sandeepraj) [inline sources/packages/web/src/components/ReadMoreContainer.vue:30]: Can you please add a comment here 
By Default the long text will be collapsed with the link option of "ReadMore "

File: sources/packages/web/tests/unit/components/ReadMoreContainer.spec.ts [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+  it("Should display expanded content when 'Read More' is clicked", async () => {
+    // Arrange
+    const expandedText = "Expanded Text";
+    // Act
+    const wrapper = mount(ReadMoreContainer, {
+      props: { expandedText },
+    });
+    await wrapper.find(toggleButtonSelector).trigger("click");
+    // Assert
```
Comment (2011sandeepraj) [inline sources/packages/web/tests/unit/components/ReadMoreContainer.spec.ts:28]: Nice!

File: sources/packages/web/tests/unit/components/ReadMoreContainer.spec.ts [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+    expect(wrapper.find(extendedContentSelector).isVisible()).toBe(true);
+    expect(wrapper.find(toggleButtonSelector).text()).toBe(expandedText);
+  });
+
+  it("Should return to initial state if toggle button is clicked twice", async () => {
+    // Arrange
```
Comment (2011sandeepraj) [inline sources/packages/web/tests/unit/components/ReadMoreContainer.spec.ts:34]: Nice again!

File: sources/packages/web/src/components/fa-application/Section.vue (hunk 1 above)
Comment (andrewsignori-aot) [reply sources/packages/web/src/components/fa-application/Section.vue:37]: The type property is used to allow the same section component to be reused in different levels of the applications, as shown below. Is it what you were asking? Please let me know.

image

**Review by 2011sandeepraj — COMMENTED** [review]
Over all Great Job. Please do look at the comments as suggestions

File: sources/packages/web/src/components/ReadMoreContainer.vue (hunk 6 above)
Comment (andrewsignori-aot) [reply sources/packages/web/src/components/ReadMoreContainer.vue:7]: You are right about the "<a>" doesn't require the href attribute, but not use it has some implications for styling and more.
An "<a>" element without href is not accessible using the tab, for instance, hence I am keeping it.
https://stackoverflow.com/questions/10510191/valid-to-use-a-anchor-tag-without-href-attribute

File: sources/packages/web/src/views/FinancialInfo.vue (hunk 4 above)
Comment (andrewsignori-aot) [reply sources/packages/web/src/views/FinancialInfo.vue:11]: Added some comments.

File: sources/packages/web/src/components/fa-application/Section.vue (hunk 3 above)
Comment (andrewsignori-aot) [reply sources/packages/web/src/components/fa-application/Section.vue:3]: Added some comments.

File: sources/packages/web/src/components/ReadMoreContainer.vue (hunk 7 above)
Comment (andrewsignori-aot) [reply sources/packages/web/src/components/ReadMoreContainer.vue:30]: Added some comments.

Comment (2011sandeepraj) [reply sources/packages/web/src/components/ReadMoreContainer.vue:30]: Where are the comments, i dont see them in the file

**Review by fwpushan — APPROVED** [review]
Good job

**Review by 2011sandeepraj — APPROVED** [review]
LGTM

**Pull request merged**